api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+
+/**
+ * Set of methods accessible to the scripts using the GoogleApps functions.
+ * @version $Id$
+ * @since 3.0
+ */
+@Component
+@Singleton
+public class GoogleAppsManagerImpl
+        implements GoogleAppsManager, EventListener, Initializable, Disposable
+{
+
+    //  -----------------------------  Lifecycle ---------------------------
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private Environment environment;
+
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> documentResolver;
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Override
+    public String getName()
+    {
+        return "googleapps.scriptservice";
+    }
+
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        log.info("GoogleAppsScriptService initting.");
+        XWiki xwiki = getXWiki();
+        XWikiContext context = xwikiContextProvider.get();
+
+        if (context != null) {
+            readConfigDoc(context);
+        }
+
+        if (xwiki != null && context != null) {
+            log.info("Initting authService.");
+            // We do not verify with the context if the plugin is active and if the license is active
+            // this will be done by the GoogleAppsAuthService and UI pages later on, when it is called within a request
+            try {
+                authService = componentManager.getInstance(GoogleAppsAuthService.class);
+                authService.setGoogleAppsManager(this);
+                xwiki.setAuthService(authService);
+                log.info("Succeeded initting authService,");
+            } catch (ComponentLookupException e) {
+                log.info("Failed initting authService", e);
+            }
+        }
+        if (authService == null) {
+            log.info("Not yet initting authService.");
+        }
+
+        try {
+            jacksonFactory = JacksonFactory.getDefaultInstance();
+            httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (Exception  e) {
+            e.printStackTrace();
+            throw new InitializationException("Trouble at initializing", e);
+        }
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        return Arrays.asList(new ApplicationReadyEvent(), new DocumentUpdatedEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        log.info("Event triggered: " + event + " with source " + source);
+        boolean applicationStarted = false, configChanged = false;
```

Comment:
For readability reasons, I would separately declare these 2 variables. They doesn't seem to be related, anyway.

Reply:
done